src/main/sensors/gyro.c

```diff
@@ -98,37 +98,46 @@ typedef struct gyroCalibration_s {
 
 bool firstArmingCalibrationWasStarted = false;
 
+typedef union gyroStage1Filter_u {
+    fastKalman_t fastKalmanFilterState[XYZ_AXIS_COUNT];
+    biquadFilter_t biquadRCFIR2FilterState[XYZ_AXIS_COUNT];
+    pt1Filter_t pt1FilterState[XYZ_AXIS_COUNT];
+} gyroStage1Filter_t;
+
 typedef union gyroSoftFilter_u {
-    biquadFilter_t gyroFilterLpfState[XYZ_AXIS_COUNT];
-    pt1Filter_t gyroFilterPt1State[XYZ_AXIS_COUNT];
-    firFilterDenoise_t gyroDenoiseState[XYZ_AXIS_COUNT];
-} gyroSoftLpfFilter_t;
+    biquadFilter_t biquadFilterState[XYZ_AXIS_COUNT];
+    pt1Filter_t pt1FilterState[XYZ_AXIS_COUNT];
+    firFilterDenoise_t denoiseFilterState[XYZ_AXIS_COUNT];
+} gyroStage2Filter_t;
 
 typedef struct gyroSensor_s {
     gyroDev_t gyroDev;
     gyroCalibration_t calibration;
-    // gyro soft filter
-    filterApplyFnPtr softLpfFilterApplyFn;
-    gyroSoftLpfFilter_t softLpfFilter;
-    filter_t *softLpfFilterPtr[XYZ_AXIS_COUNT];
+
+    // stage1 gyro soft filter (pt1, fkf, biquad rc+fir2)
+    filterApplyFnPtr stage1FilterApplyFn;
+    gyroStage1Filter_t stage1Filter;
+    filter_t *stage1FilterPtr[XYZ_AXIS_COUNT];
+
+    // lagged moving average smoothing
+    filterApplyFnPtr lmaSmoothingApplyFn;
```

Comment:
It's a bit strange to have poiter dispatch to single function ...

Reply:
@ledvinap AFAIk this is so we can do `nullFilterApply` later, i.e. conditionally disable.

```
gyroSensor->lmaSmoothingApplyFn = nullFilterApply;
```